cg/meta/backup/pdc.py

```diff
@@ -41,7 +61,63 @@ def retrieve_file_from_pdc(self, file_path: str, target_path: str = None) -> Non
         self.run_dsmc_command(command=command)
 
     def run_dsmc_command(self, command: list) -> None:
-        """Runs a DSMC command"""
+        """Runs a DSMC command.
+        Raises:
+            PdcError when unable to process command.
+        """
         LOG.debug("Starting DSMC command:")
         LOG.debug(f"{self.process.binary} {' '.join(command)}")
-        self.process.run_command(parameters=command, dry_run=self.dry_run)
+        try:
+            self.process.run_command(parameters=command, dry_run=self.dry_run)
+        except Exception as error:
+            raise PdcError(f"{error}") from error
+
+    def validate_flow_cell_backup_possible(
+        self, db_flow_cell: Flowcell, flow_cell_encryption_api: FlowCellEncryptionAPI
+    ) -> None:
+        """Check if back-up of flow cell is possible.
+        Raises: DcmsAlreadyRunningError if there is already a Dcms process ongoing"""
+        if self.is_dcms_running():
+            exit(0)
+        if db_flow_cell and db_flow_cell.has_backup:
+            raise FlowCellAlreadyBackeupError(
+                f"Flow cell: {db_flow_cell.name} is already backed-up"
+            )
+        if not flow_cell_encryption_api.complete_file_path.exists():
+            raise FlowCellEncryptionError(
+                f"Flow cell: {flow_cell_encryption_api.flow_cell.id} encryption process is not complete"
+            )
+
+    def backup_flow_cell(
+        self, files_to_archive: list[Path], status_db: Store, db_flow_cell: Flowcell
+    ) -> None:
+        """Back-up flow cell files."""
+        archived_file_count: int = 0
+        for encrypted_file in files_to_archive:
+            try:
+                self.archive_file_to_pdc(file_path=encrypted_file.as_posix())
+                archived_file_count += 1
+            except PdcError:
+                LOG.debug(f"{encrypted_file.as_posix()} cannot be archived")
+            if archived_file_count == len(files_to_archive) and not self.dry_run:
+                status_db.update_flow_cell_has_backup(flow_cell=db_flow_cell, has_backup=True)
+                LOG.debug(f"Flow cell: {db_flow_cell.name} has been backed up")
```

Comment:
I think this merits INFO
```suggestion
                LOG.info(f"Flow cell: {db_flow_cell.name} has been backed up")
```